io-engine/src/grpc/v1/pool.rs

```
@@ -57,6 +68,85 @@ impl From<GrowPoolRequest> for FindPoolArgs {
     }
 }
 
+/// Parse and return encryption parameters from a Kubernetes secret or a file.
+async fn _params_get_from_secret_source(secret_source_name: &str) -> Result<PoolEncKey, Status> {
+    let secret_params = match current_platform_type() {
+        PlatformType::K8s => {
+            let client = kube::Client::try_default()
+                .await
+                .expect("Should be able to create kube client");
```

Should return an error

---

> Should return an error

done